### PR TITLE
ESQL: `MV_FIRST` and `MV_LAST`

### PR DESCRIPTION
**Pull request by nik9000**
This creates the `MV_FIRST` and `MV_LAST` functions that return the first and last values from a multivalue field. They are noops from a single valued field. They are quite similar to `MV_MIN` and `MV_MAX` except they work on positional data rather than relative size. That sounds like a large distinction, but in practice our multivalued fields are often sorted. And when they operate on sorted arrays `MV_MIN` does *the same* thing as `MV_FIRST`.

But there are some cases where it really does matter - say you are `SPLIT`ing something - so `MV_FIRST(SPLIT("foo;bar;baz", ";"))` gets you `foo` like you'd expect. No sorting needed.

Relates to #103879

### CODE REVIEW

**Comment by nik9000** [comment]
I've added the ESQL-ui tag because this is a new function. I'm not sure you actually need to be notified about this but I figured better safe than sorry.

File: x-pack/plugin/esql/compute/ann/src/main/java/org/elasticsearch/compute/ann/MvEvaluator.java [hunk 1]
```diff
@@ -16,17 +16,30 @@
...
+ *     Pairwise processing is <strong>generally</strong> simpler and looks
+ *     like {@code int process(int current, int next)}. Use it when the result
+ *     is a primitive.</p>
+ * <p>
+ *     Accumulator processing is a bit more complex and looks like
+ *     {@code void process(State state, int v)} and it useful when you need to
+ *     accumulate more data than fits in a primitive result. Think Kahan summation.
+ * </p>
+ * <p>
+ *     Position at a time processing just hands the block, start index, and end index
```
Comment (bpintea) [inline x-pack/plugin/esql/compute/ann/src/main/java/org/elasticsearch/compute/ann/MvEvaluator.java:36]: Nice. I think this should make a MV function returning the element at a position easy now.

File: docs/reference/esql/functions/mv_last.asciidoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/string.csv-spec[tag=mv_last-result]
+|===
+
+The order that <<esql-multivalued-fields, multivalued fields>> are read from underlying storage is not
+guaranteed. It is *frequently* ascending, but don't rely on that. If you need the maximum field value
+use <<esql-mv_max>> instead of `MV_LAST`. `MV_MAX` has optimizations for sorted values so there isn't
+a performance benefit to `MV_LAST`. `MV_LAST` is really only useful with functions that create
```
Comment (bpintea) [inline docs/reference/esql/functions/mv_last.asciidoc:22]: I'd tone the "really only" down ("mostly"?). I can imagine indexed MVs keyword fields and wanting to get the 1st/last value.

**Review by bpintea — APPROVED** [review]
LGTM.

File: docs/reference/esql/functions/mv_last.asciidoc (hunk 2 above)
Comment (nik9000) [reply docs/reference/esql/functions/mv_last.asciidoc:22]: Sure

File: x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/Clusters.java [hunk 3]
```diff
@@ -1,4 +1,7 @@
 /*
+Viewed
+File renamed without changes.
+
```
[the remark this answers is not in the document]
Comment (nik9000) [reply x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/Clusters.java:4]: Weird. Yeah.

**Comment by nik9000** [comment]
retest this please

**Comment by nik9000** [comment]
Oh exciting! the github burbles this morning seem to have broken the CI linkage so this was merged without running all the tests. I mean, they passed locally but this could get exciting....

**Pull request merged**